src/classes/ADDR_GoogleGeoAPI_Validator.cls

```diff
@@ -310,29 +130,368 @@ public with sharing class ADDR_GoogleGeoAPI_Validator implements ADDR_IValidator
     }
-    
+
     /*******************************************************************************************************
     * @description Apex model of the viewport element of the Google JSON response.
     */
     public class Viewport {
         public Location northeast;
         public Location southwest;
     }
-    
+
     /*******************************************************************************************************
     * @description Parses a JSON strings and returns a GoogleGeoResponse Apex object.
-    * @param String The JSON string.
+    * @param json The JSON string.
     * @return GoogleGeoResponse The Apex object the JSON string has been transformed into.
     */
     public static GoogleGeoResponse parse(String json) {
         return (GoogleGeoResponse) System.JSON.deserialize(json, GoogleGeoResponse.class);
     }
+
+    /**
+    * @description Address Verification Wrapper Class
+    */
+    public class AddressVerifier {
+
+        private Address__c address;
+        private String endPoint;
+        private Addr_Verification_Settings__c settings;
+        private GoogleGeoResponse googleResponse;
+        private HttpResponse rawResponse;
+
+        /** @description SubPremise from the original address to append to the Street Address (if not null) */
+        private String subPremise;
+
+        /** @description Constructor with the address to verify */
+        public AddressVerifier(Address__c addr) {
+            this.address = addr;
+        }
+
+        /**
+        * @description Pass an instance of the AddressVerificationSettings object
+        * @return Instance
+        * */
+        public AddressVerifier withAddressSettings(Addr_Verification_Settings__c settings) {
+            this.settings = settings;
+            return this;
+        }
+
+        /**
+        * @description Pass the endpoint to use
+        * @return Instance
+        */
+        public AddressVerifier withEndPoint(String endPoint) {
+            this.endPoint = endPoint;
+            return this;
+        }
+
+        /**
+       * @description Verify this Aaddress record and return an instance of this class
+       * @return Instance
+       */
+        public AddressVerifier verifyAddress() {
+
+            if (settings == null || settings.Auth_Token__c == null || endPoint == null) {
+                address.Verified__c = false;
+                address.API_Response__c = 'MISSING AUTHID OR ENDPOINT.  Please see the Address Verification menu in the NPSP Settings tab for more information';
+                return this;
+            }
+
+            if (!calloutToAddressService()) {
+                return this;
+            }
+
+            if (googleResponse.status != ADDR_Validator_UTIL.RESPONSE_OK || googleResponse.results == null) {
+                handleOtherResponses();
+                return this;
+            }
+
+            Boolean ignoreAmbiguous = settings.Reject_Ambiguous_Addresses__c == true;
```

Comment:
It took me a while to understand this part of the code due to `Reject_Ambiguous_Addresses__c` (mis-aligned field API name and Help Text)... if it is set and the address is the partial match, it should mark the address as `Ambiguous__c` and 'Verified__c`. 

I tested few address verifications where the `Reject_Ambiguous_Addresses__c`  is unchecked. 
- One test was verification of an address with Zip Code, State and Country (US). The address verifier response contains `"partial_match" : true`. The address was marked as not `Ambiguous__c` and set as 'Verified__c` after the verification.
- The other test was an address containing format something like "100-1230 Water St, ..., Canada" then the address verification returns address as "100 Water St,...", ie the street number is removed while the apt number is taken as the street number. This does not seem correct, but I have not verified this against `master` to see if it is an existing issue.
- The the address format is something like  "1230 Water St, Apt 100, .., Canada." , the `"partial_match" : true` is returned, I thought it should be false.

If the `Reject_Ambiguous_Addresses__c` is checked, then addresses containing the second and third format above would be marked as ambiguous because the `partial_match` is true.  Is this expected?

Reply:
I went through the original code and I cannot find where the new logic is any different than the previous logic. Do you still have an org built from master configured for address verification? If not, I'll build one to compare results just to make sure this is unchanged.

Reply:
The code with regards to setting the Ambigous__c is fine. The issue was reading it and understanding the result. Also, the mentioned address format examples (Canadian address) return different API response in the branch org and the master org. I expected the `partial_match` to be false when the address is exact. However, the same address format in the `master` org returns "bad request" response so cannot compare.
In summary, the question boils down to different API response for the same address format when running the verification on the branch org compared to the `master` org. The WI contains screenshots and more detail.